scene/2d/canvas_item.cpp

```diff
@@ -350,31 +350,19 @@ Transform2D CanvasItem::_edit_get_transform() const {
 #endif
 
 bool CanvasItem::is_visible_in_tree() const {
-	if (!is_inside_tree()) {
-		return false;
-	}
-
-	const CanvasItem *p = this;
-
-	while (p) {
-		if (!p->visible) {
-			return false;
-		}
-		p = p->get_parent_item();
-	}
-
-	return true;
+	return visible && visible_in_tree;
 }
```

Comment:
Why is the new variable called `visible_in_tree`, if it is not whether the item is visible in the tree? Should this be `parent_visible_in_tree`?

Reply:
In 4.0 it can depend on Window, which might not be a direct parent.

Reply:
Does `ancestors_visible_in_tree` make sense?

Reply:
FYI, the classref uses [antecedents](https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#class-canvasitem-method-is-visible-in-tree).

> if the node is present in the SceneTree, its visible property is true and all its antecedents are also visible

Reply:
I'm not sure about antecedent - it is not really a commonly understand English word, and most people would have to look it up to find the definition (including me as a native speaker :grinning: ).

Reply:
Either way, changing `is_visible_in_tree()` would break compatibility in `3.x`.

Reply:
Then `parent_visible_in_tree` is what it tracks I think? There's no `Window` visibility involved in `3.x`.